challenges/challenges.py

```diff
@@ -166,69 +141,41 @@ async def battle_check(self):
                                 win = False
                             if "rank" in battle['battle'] and battle['battle']['mode'] != "soloShowdown" and battle['battle']['rank'] > 2:
                                 win = False
-                            if battle['battle']['mode'].lower().replace('-', '').replace(' ', '') in ('roborumble', 'biggame'):
-                                continue
-                            brawler_name = player['brawler']['name']
-                            if group_plant:
-                                if brawler_name in ("SPIKE", "ROSA", "SPROUT"):
-                                    if win:
-                                        progress += 1
-                                        if brawler_name in wins:
-                                            wins[brawler_name] += 1
-                                        else:
-                                            wins[brawler_name] = 1
-                                    else:
-                                        if brawler_name in loses:
-                                            loses[brawler_name] += 1
-                                        else:
-                                            loses[brawler_name] = 1
+
+                            streak = await self.config.members(user).streak()
+                            if win and player['brawler']['trophies'] >= 500 and battle['battle']['mode'] in ('brawlBall', 'gemGrab', 'bounty', 'siege', 'hotZone'):
```

Comment:
Okay, Bob gave me some evil reqs for here. Basically, your streak is only valid if you got it with a brawler above 500 and in a 3v3 non-special mode. But. If you lose with a <500 brawler in any mode between those streaks, your streak is forfeited. I tried to implement it as well as I could, if it sounds confusing, well, it is.